client.go

```diff
@@ -182,25 +181,29 @@ func (c *Client) sendQuery(ctx context.Context, q *Query) (httpext.Response, err
 	return *response, err
 }
 
-func (c *Client) Push(ctx context.Context) (httpext.Response, error) {
+func (c *Client) Push() (httpext.Response, error) {
 	// 5 streams per batch
 	// batch size between 800KB and 1MB
-	return c.PushParameterized(ctx, 5, 800*1024, 1024*1024)
+	return c.PushParameterized(5, 800*1024, 1024*1024)
 }
 
 // PushParametrized is deprecated in favor or PushParameterized
-func (c *Client) PushParametrized(ctx context.Context, streams, minBatchSize, maxBatchSize int) (httpext.Response, error) {
-	c.logger.Warn("method pushParametrized() is deprecated and will be removed in future releases; please use pushParameterized() instead")
-	return c.PushParameterized(ctx, streams, minBatchSize, maxBatchSize)
+func (c *Client) PushParametrized(streams, minBatchSize, maxBatchSize int) (httpext.Response, error) {
+	if state := c.vu.State(); state == nil {
+		return *httpext.NewResponse(), errors.New("state is nil")
+	} else {
+		state.Logger.Warn("method pushParametrized() is deprecated and will be removed in future releases; please use pushParameterized() instead")
+	}
+	return c.PushParameterized(streams, minBatchSize, maxBatchSize)
 }
 
-func (c *Client) PushParameterized(ctx context.Context, streams, minBatchSize, maxBatchSize int) (httpext.Response, error) {
-	batch := newBatch(ctx, c.cfg.Labels, streams, minBatchSize, maxBatchSize)
-	return c.pushBatch(ctx, batch)
+func (c *Client) PushParameterized(streams, minBatchSize, maxBatchSize int) (httpext.Response, error) {
+	batch := newBatch(c.vu.Context(), c.vu.State(), c.cfg.Labels, streams, minBatchSize, maxBatchSize)
```

Comment:
I think we need to check for vu.State() here as well:

```suggestion
    if state := c.vu.State(); state == nil {
		return *httpext.NewResponse(), errors.New("state is nil")
	}
	batch := newBatch(c.vu.Context(), c.vu.State(), c.cfg.Labels, streams, minBatchSize, maxBatchSize)
```

Reply:
I also wondered but it's already done in `pushBatch` so ... I don't think it's needed, but I can add it :shrug:

Reply:
Do you want me to add it so we can merge this ?

Reply:
`PushParameterized` is the function that is usually called from within the JS test file. So I'd add the check there as well.

Reply:
fixed in [77cd232](https://github.com/grafana/xk6-loki/pull/28/commits/77cd23223811a68a7f3025bf832bcc549755ec5d)